pkg/sql/catalog/descriptor.go

```diff
@@ -276,107 +393,282 @@ type TableDescriptor interface {
 	// canonical order, see Index.Ordinal().
 	FindIndexWithName(name string) (Index, error)
 
+	// GetNextIndexID returns the next unused index ID for the table. Index IDs
+	// are unique within a table, but not globally.
 	GetNextIndexID() descpb.IndexID
 
+	// HasPrimaryKey returns true if the table has a primary key. This will be
+	// true except for in a transaction where a user has dropped a primary key
+	// in preparation to add a new one. In CockroachDB, all tables have primary
+	// keys, even if they're not defined by the user.
 	HasPrimaryKey() bool
+	// PrimaryKeyString returns the pretty-printed primary key declaration for a
+	// table descriptor.
```

Comment:
Unrelated to your changes but this jumped at me as something that doesn't seem to belong in this interface at all. I'll follow up on this.

Reply:
Please take a look at https://github.com/cockroachdb/cockroach/pull/71082